security/advancedtls/crl_provider.go

```diff
@@ -52,6 +52,9 @@ type CRLProvider interface {
 	//
 	// [RFC5280 - Undetermined]: https://datatracker.ietf.org/doc/html/rfc5280#section-6.3.3
 	CRL(cert *x509.Certificate) (*CRL, error)
+
+	// Close cleans up resources allocated by the provider.
+	Close()
```

Comment:
Why change this?  This forces the static CRL provider to implement it, when it doesn't need it.

Reply:
I was taking another look at gRFC and realized that `Close()` is specified at the interface level there - https://github.com/grpc/proposal/pull/382/#discussion_r1379510339
Then I checked the Go repo and realized that having `Close()' at interface level plus no-op impls (if needed) is an existing pattern - https://github.com/grpc/grpc-go/blob/70f1a4045da95b93f73b6dbdd7049f3f053c0680/balancer/balancer.go#L388
During internal discussion folks suggested to make that change - WDYT?

Reply:
We need to close the LB policy from gRPC; hence it's included in that interface.  You don't close the CRL provider from advancedTLS.  So the same reasoning doesn't apply in this case.  I would not add it here, since it isn't necessary.

Reply:
This was designed to be very close to the Credential Reloading feature which has `Close` in the interface
https://github.com/grpc/grpc-go/blob/6bed35367c88ff9587572ce58746beac6fafb612/credentials/tls/certprovider/provider.go#L75-L89

Reply:
If the cert provider doesn't need it (i.e. gRPC never calls it), then we should consider removing it from there.  I'm not sure whether that's the case. 
 But that's not justification to add it here.  It's not needed here, so we probably shouldn't add it.

Feel free to disregard my advice if you really want this.

Reply:
I agree with Doug here, for example `Close()` from provider.go is used in [xDS case](https://github.com/grpc/grpc-go/blob/482de22249427eca78db31ebb9937fbd30e81599/xds/internal/balancer/cdsbalancer/cdsbalancer.go#L182) by gRPC for recreation.
Since we don't have such cases for CRL, I'd prefer to drop it from the interface. @gtcooke94 WDYT?